### PR TITLE
Renamed struct item_t to battle_item_t

### PR DESCRIPTION
**Pull request by Elsa-Athiley**
Because of duplicate definitions between some battle code structs and game-state structs we had to change the naming of our structs. I have changed the item_t struct to battle_item_t and updated all instances of the struct in each file. It builds correctly and our example files still work correctly.

### CODE REVIEW

**Review by adinilfeld — COMMENTED** [review]
Looks good to me, the only thing I'd add is that we should probably update [this wiki page](https://github.com/uchicago-cs/chiventure/wiki/RPG-Battle-Systems:-Design-Document) with the new terminology once this is merged.

Also, are there any comments in the code mentioning `item_t` that should be changed to `battle_item_t`? Just want to make sure that there's not any confusion regarding which item they're referring to.

File: include/battle/battle_logic.h [hunk 1]
```diff
@@ -46,7 +46,7 @@ turn_t goes_first(battle_t *b);
  * Returns:
  *    Returns a pointer to the desired item 
  */
-item_t *find_item(item_t *inventory, int id);
+battle_item_t *find_item(battle_item_t *inventory, int id);
```
Comment (firatciftci) [inline include/battle/battle_logic.h:49]: The name of this function should probably be changed to `find_battle_item` – this also applies to all other functions with similar names.

File: include/battle/battle_logic.h [hunk 2]
```diff
@@ -56,7 +56,7 @@ item_t *find_item(item_t *inventory, int id);
  * Returns:
  *  Always returns 0
  */
-int consume_item(combatant_t *c, item_t *item);
+int consume_item(combatant_t *c, battle_item_t *item);
```
Comment (firatciftci) [inline include/battle/battle_logic.h:59]: The name of this function should probably be changed to `consume_battle_item`.

File: include/battle/battle_logic.h [hunk 3]
```diff
@@ -46,7 +46,7 @@ turn_t goes_first(battle_t *b);
  * Returns:
  *    Returns a pointer to the desired item
```
Comment (firatciftci) [inline include/battle/battle_logic.h:47]: We should probably state `... desired battle item` instead.

File: include/battle/battle_logic.h [hunk 4]
```diff
@@ -46,7 +46,7 @@ turn_t goes_first(battle_t *b);
  * Returns:
  *    Returns a pointer to the desired item 
  */
-item_t *find_item(item_t *inventory, int id);
+battle_item_t *find_item(battle_item_t *inventory, int id);
 
 /*
  * Consumes an item for the said combatant
```
Comment (firatciftci) [inline include/battle/battle_logic.h:52]: We should probably state `Consumes a battle item ...` instead.

File: src/battle/examples/battle_command_example.c [hunk 5]
```diff
@@ -106,6 +107,26 @@ int print_moves(chiventure_ctx_battle_t *ctx)
     return SUCCESS;
 }
 
+/* Prints out the avaliable items for the player
+ * Parameter:
+ *  ctx: the main structure of the game
+ * Returns:
+ *  Always SUCCESS
+ */ 
+int print_items(chiventure_ctx_battle_t *ctx)
```
Comment (firatciftci) [inline src/battle/examples/battle_command_example.c:116]: The name of this function should probably be changed to `print_battle_items`.

File: src/battle/examples/battle_command_example.c [hunk 6]
```diff
@@ -106,6 +107,26 @@ int print_moves(chiventure_ctx_battle_t *ctx)
     return SUCCESS;
 }
 
+/* Prints out the avaliable items for the player
```
Comment (firatciftci) [inline src/battle/examples/battle_command_example.c:110]: We should probably state `... available battle items ...` instead.

File: src/battle/examples/battle_command_example.c [hunk 7]
```diff
@@ -166,9 +187,42 @@ int read_move(char **args, chiventure_ctx_battle_t *ctx)
         // prints out possible commands for the user to use
         printf("Here are the possible commands!\n");
         printf("MOVE USE <move_name> ON <enemy_name>\n");
-        printf("MOVE INFO <move_name>\n\n");
+        printf("MOVE INFO <move_name>\n");
+        printf("USE <item_id>\n\n");
+        printf("Here is the list of available items!\n");
```
Comment (firatciftci) [inline src/battle/examples/battle_command_example.c:192]: We should probably state `... available battle items!\n` instead.

File: src/battle/examples/battle_command_example.c [hunk 8]
```diff
@@ -166,9 +187,42 @@ int read_move(char **args, chiventure_ctx_battle_t *ctx)
...
     }
+    //   // handles the command USE <item>
+     else if (strncmp(args[0], "USE", MAX_COMMAND_LENGTH) == 0) 
+    {
+        
+        battle_item_t *item = find_item(ctx->game->battle->player->items, atoi(args[1]));
+        printf("Determined command as USE %s\n\n", item->name);
+        if (item == NULL)
+        {
+            printf("Couldn't find the item you were looking for!\n");
```
Comment (firatciftci) [inline src/battle/examples/battle_command_example.c:205]: We should probably state `... the battle item` instead.

File: src/battle/examples/battle_command_example.c [hunk 9]
```diff
@@ -166,9 +187,42 @@ int read_move(char **args, chiventure_ctx_battle_t *ctx)
...
+            return FAILURE;
+        }
+        if (item->quantity == 0)
+        {
+            printf("Sorry you don't have any more of that item!\n");
```
Comment (firatciftci) [inline src/battle/examples/battle_command_example.c:210]: We should probably state `... that battle item` instead.

File: src/battle/examples/battle_command_example.c [hunk 10]
```diff
@@ -166,9 +187,42 @@ int read_move(char **args, chiventure_ctx_battle_t *ctx)
...
+            return FAILURE;
+        }
+
+        res = use_item(ctx->game->battle->player, atoi(args[1]));
```
Comment (firatciftci) [inline src/battle/examples/battle_command_example.c:214]: Couldn't comment on the actual file for this function, so adding it here – the name of this function should probably be changed to `use_battle_item`.

**Review by firatciftci — CHANGES_REQUESTED** [review]
I have requested some changes which basically extend the renaming of the notion of an `item` to a `battle item` in more locations of the `battle` codebase. Also, as @adinilfeld has mentioned, the `battle` team should also update their respective wiki page to reflect this substantial change in names.

**Comment by firatciftci** [comment]
Also, it seems that the automated tests for this PR is now failing due to another PR from the `battles` team that was merged earlier today. I suggest looking over all of the new modifications to the `battles` module and update this PR accordingly so that the tests pass again. If you'd like me to go over/help with anything in this regard, feel free to let me know!

**Comment by Elsa-Athiley** [comment]
Applied all of firats suggestions. The automated tests were failing because of a new module added from another Pull request that was still using the old name of item and not battle_item. All test should pass now. However there is another pull request out #905 that hasnt been merged yet that uses the old naming conventions. Depending on which PR gets merged first additional adjustments may need to be made in this PR or the other one.

**Comment by Elsa-Athiley** [comment]
Fixed the merge conflicts that were a result of my other Pull request going through that had some changes to the `battle_command_example.c` file.

**Review by adinilfeld — APPROVED** [review]
Looks great!

**Review by firatciftci — APPROVED** [review]
All of the changes look good. Great work!

**Comment by Elsa-Athiley** [comment]
Closing Summary: All instances of item_t have now been changed to battle_item_t in all files in the src/battle, include/battle, and test/battle directories.

**Comment by firatciftci** [comment]
Issue Score: ✔️++

Comments:
Great job!

**Pull request merged**